common/k8stest/util_pool.go

```diff
@@ -121,3 +130,131 @@ func GetCapacityAndAllocatedAllSnapshotsAllReplicasOnPool(poolName string) (int6
+// by creating the configmap before this function is invoked
+func CreateDiskPoolsConfiguration() error {
+	// weights for sorting device links in order of preference
+	var devLinkWeights = map[string]int{
+		"by-uuid":     0,
+		"by-id":       1,
+		"by-partuuid": 2,
+		"by-path":     3,
+	}
+	// struct to de-serialise plugin output for block devices
+	type msBlockDev struct {
+		Available bool     `json:"available"`
+		DevLinks  []string `json:"devlinks"`
+		DevName   string   `json:"devname"`
+		DevPath   string   `json:"devpath"`
+	}
+	configMapApi := gTestEnv.KubeInt.CoreV1().ConfigMaps(common.NSDefault)
+	cmLst, err := configMapApi.List(context.TODO(), metaV1.ListOptions{})
+
+	if err == nil {
+		// if the configmap already exists the use it as is
+		for _, cm := range cmLst.Items {
+			if cm.Name == configMapName {
+				log.Log.Info("CreateDiskPoolsConfiguration: using existing config map", "name", cm.Name)
+				return nil
+			}
+		}
+		// compile a list of devices on each node.
+		nodes, err := GetMayastorNodeNames()
+		if err != nil {
+			log.Log.Info("CreateDiskPoolsConfiguration: GetMayastorNodeName", "error", err)
+			return err
+		}
+		cmapData := make(map[string]string)
+		for _, node := range nodes {
+			// retrieve block devices on a node in json format
+			cmd := mcpV1.GetMayastorPluginCmd("-n", common.NSMayastor(), "get", "block-devices", node, "-o", "json")
+			var out bytes.Buffer
+			log.Log.Info("About to execute:", "cmd", cmd)
+			cmd.Stdout = &out
+			err = cmd.Run()
+			if err != nil {
+				log.Log.Info("CreateDiskPoolsConfiguration:", "cmd", cmd, "error", err)
+				return err
+			}
+			log.Log.Info("CreateDiskPoolsConfiguration:", "raw data", out.String())
+			// de-serialise the json data
+			var data []msBlockDev
+			err = json.Unmarshal(out.Bytes(), &data)
+			if err != nil {
+				log.Log.Info("CreateDiskPoolsConfiguration: json Unmarshal", "error", err)
+				return err
+			}
+			var disks []string
+			for _, bd := range data {
+				if !bd.Available {
+					// disk is not available for use by mayastor
+					continue
+				}
+				// ignore loop devices
+				if strings.HasPrefix(bd.DevName, "/dev/loop") {
+					continue
+				}
+				// default to device name
+				device := bd.DevName
+				if len(bd.DevLinks) != 0 {
+					// if a list device links exists then use that as the device
+					// sort in order of preference and use the first one.
+					sort.Slice(bd.DevLinks, func(i int, j int) bool {
+						keyI := strings.Split(bd.DevLinks[i], "/")[3]
+						keyJ := strings.Split(bd.DevLinks[j], "/")[3]
+						weightI := 10
+						if val, ok := devLinkWeights[keyI]; ok {
+							weightI = val
+						}
+						weightJ := 10
+						if val, ok := devLinkWeights[keyJ]; ok {
+							weightJ = val
+						}
+						if weightI == weightJ {
+							// if weights are the same sort alphabetically for consitent
+							// results
+							return bd.DevLinks[i] < bd.DevLinks[j]
```

Comment:
Return type is error . But here it is not?

Reply:
this is a return statement for the sorting function declared on line 206, so must be bool